kivy/uix/screenmanager.py

```diff
@@ -897,34 +897,46 @@ def remove_widget(self, *l):
         screen = l[0]
         if not isinstance(screen, Screen):
             raise ScreenManagerException(
-                'ScreenManager uses remove_widget only to remove' +
-                'screens added via add_widget! use real_remove_widget.')
+                'ScreenManager uses remove_widget only for removing Screens.')
 
         if not screen in self.screens:
             return
+
         if self.current_screen == screen:
             other = next(self)
-            if other:
+            if screen.name == other:
+                self.current = None
+                screen.parent.real_remove_widget(screen)
+            else:
                 self.current = other
-        
-        # ensure screen is removed from it's previous parent
-        if screen.parent:
-            screen.parent.real_remove_widget(screen)
-            
+
         screen.manager = None
         screen.unbind(name=self._screen_name_changed)
         self.screens.remove(screen)
 
-    def real_add_widget(self, *l):
-        super(ScreenManager, self).add_widget(*l)
+    def clear_widgets(self, screens=None):
+        if not screens:
+            screens = self.screens
+        remove_widget = self.remove_widget
+        for screen in screens:
+            remove_widget(screen)
 
-    def real_remove_widget(self, *l):
-        super(ScreenManager, self).remove_widget(*l)
+    def real_add_widget(self, screen, *args):
+        # ensure screen is removed from its previous parent
+        if screen.parent:
+            screen.parent.real_remove_widget(screen)
```

Comment:
Since Screen can be added to any `Widget` not just `ScreenManager`, it's probably better to check if `screen.parent` is a `ScreenManager` instance and call `remove_widget` if it's not the case.